enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,446 @@
+
+At this time, the current sentiment for 4.x is to opt in to receiving entitlements instead of the default being
+the credentials are "just there", where either an annotation (or perhaps new API field(s) in the case of builds).  
+
+As noted in the open question, simply getting credentials at the global level may not be sufficient, so additional
+alternatives / complementary pieces to a global solution are articulated below.
+
+The preferred implementation path at this time is to build the following set of components inject the global secret contents 
+in a pod:
+ - a [CSI plugin](https://github.com/container-storage-interface/spec) would fetch the secret(s) and write it on the underlying 
+ node/host filesystem on a path we control so that CRIO can mount the content
+ - this CSI plugin essentially provides a new "entitlement" Volume type 
+ - it is a special case type that is not intended to be the more generic CSI Volume type that is already present
+ upstream
+ - and while it is similar to a HostPath volume, it won't appear in the pod as a HostPath, with the requisite 
+ permissions that implies
+ - the CSI plugin would be hosted as a `DaemonSet` on the k8s cluster, to get per node/host granularity
+ - and that `DaemonSet` would be privileged so as to have access to the pod's associated node/host file system.
+ - the [CSI spec](https://github.com/container-storage-interface/spec/blob/master/spec.md) details the entire architecture,
+ concepts, and protocol
+ - but to try and summarize a key element for us: CSI has two step mounting process. staging and publishing. Staging is 
+ node local and publish is pod local. So we will probably want to stage the content once and publish it for each pod 
+ (which is nothing but creating a bind mount for original volume)
+ - That means we do not have to necessarily read and copy the secret(s) contents on every pod creation
+ - We can build an associated controller that watches the global secret, or per namespace secrets that have been 
+ annotated/labelled, and if possible that somehow engages the CSI plugin to redo the staging step.  This is *somewhat*
+ akin to the injection controller employed by the Global Proxy support.
+ - Moving on from the CSI plugin, an admission controller/webhook path (see open questions regarding "packaging" options there) 
+ that based on the presence of the annotation on the pod (or one of several for per namespace perhaps), and the presence 
+ of a global whitelist that stipulates which namespaces can annotate their pod.  See risks and mitigations for 
+ speculation around the need for more granular control.
+ - can mutate the pod, adding volume mounts of the "entitlement" volume 
+
+###### Build Specific consumption path
+
+Once the "entitlement" volume is mounted in the build pod, the openshift builder image will need to map the pod mounted volumes to the 
+requisite buildah parameters (much like it does today for existing secrets/configmaps mounted into the build pod).
```

Comment:
it doesn't do this today.  we do not setup any buildah mounts today.

configmap+secret content is mounted into the build pod, but it's supplied to buildah as part of the buildcontextdir, just like the source.

we have a *goal* to change that consumption model to be mount-based, but it's not today.

Reply:
ok just deleting `(much like it does today for existing secrets/configmaps mounted into the build pod).` in the next push

the other piece I thought I copy/pasted from an earlier comment from you.

Reply:
i think my original comment was talking about a need to have the the build controller add mounts to the pod (as it does today for configmaps+secrets).  Which now the buildcontroller will not do since csi+admission will do it.

Reply:
cool  ... hopefully the new version I pushed yesterday better reflects that  ... let me know what you think when you get the chance